src/CommunityToolkit/Xamarin.CommunityToolkit/Helpers/LocalizedString.shared.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.ComponentModel;
+
+namespace Xamarin.CommunityToolkit.Helpers
+{
+#if !NETSTANDARD1_0
+	public class LocalizedString : INotifyPropertyChanged
+	{
+		public LocalizedString(Func<string> generator = null)
+            : this(LocalizationResourceManager.Current, generator)
+        {
+        }
+
+		public LocalizedString(LocalizationResourceManager localizationManager, Func<string> generator = null)
+		{
+			Generator = generator;
+			localizationManager.PropertyChanged += (sender, args) => Invalidate();
+		}
+
+		Func<string> generator;
+
+		public Func<string> Generator
+		{
+			get => generator;
+			set
+			{
+				generator = value;
+				Invalidate();
+			}
+		}
+
+		public string Localized => Generator?.Invoke();
+
+		public static implicit operator LocalizedString(Func<string> func) => new LocalizedString(func);
```

Comment:
Nica catch! But the linker could remove it, so we need to alert the linker.

```suggestion
        [Xamarin.Forms.Internals.Preserve(Conditional = true)]
		public static implicit operator LocalizedString(Func<string> func) => new LocalizedString(func);
```

Reply:
Unfortunately it's not very useful since lambda expressions have no type, and `new(()=>{})` looks better then `(Func<string>)()=>{}`, but maybe this will be useful to someone.

Thanks for `Preserve`! I'll add it

Reply:
> Unfortunately it's not very useful since lambda expressions have no type

For that, you can create an overload with an `Action`, o make it more generic. I need to play a little bit more with this Generator Func to see the usage and came with better (or bad) ideas.

```csharp
class LocalizedString<T>
{
     public Func<T> Generator { get ; set ; } 
}
```

Reply:
Lambda expressions don't have type, so it's impossible to write implicit converting from them. But it's still might be useful if someone will cast manually, or would have method that generates the string.

- Action doesn't make sense here, since the whole point of generator is to return value that will be displayed on the page, and Actions don't return value. Also, you'll still need to explicitly cast lambda expression to action for implicit conversion to work, so there is no difference here.
- Making class generic wouldn't help with implicitly casting lambda expressions, but would allow more methods to be converted to `LocalizedString`. Page will call `ToString` on the result anyway, but I think point of `LocalizedString` is to be a string, so it doesn't need to be generic. Could you provide some use case for generic `LocalizedString`?